independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/CreationalContextImpl.java

```diff
@@ -129,4 +133,50 @@ public static <I> void addDependencyToParent(InjectableBean<I> bean, I instance,
         }
     }
 
+    static <T> InjectionPoint getCurrentInjectionPoint(CreationalContext<T> ctx) {
+        CreationalContextImpl<?> instance = unwrap(ctx);
+        while (instance != null) {
+            synchronized (instance) {
```

Comment:
Do I understand it correctly that the synchronized construct is here just because of a "creative" usage of the `BeanManager` API? I seriously doubt that it would help. I mean thread-safety is one thing but correct result is a different thing.

Reply:
Yea, I bothered Lada with possible cases that might break this earlier, see https://github.com/quarkusio/quarkus/pull/39837#discussion_r1551861539